File: problem_set/python/dec2bin.py
'''FILE    : Hex Oct Bin to Dec
   WRITER  : NAT KANJANASIRI
   DATE    : OCT 16, 2020
   PURPOSE : Convert base 2, 8, 16 to base 10'''
number = int(input("Enter number : "))
obase = ''
while(number > 0):        
        obase = str(number % 2) + obase
        number = number // 2
print(obase)

File: problem_set/c/dec2bin.c
/* FILE    : Dec to Hex Oct Bin
 * WRITER  : NAT KANJANASIRI
 * DATE    : FEB 1, 2023
 * PURPOSE : Convert base 10 to 2, 8, 16
 */
#include <stdio.h>
#include <string.h>

int main(void){
	char table[16] = "0123456789ABCDEF";
	char buffer[30], answer[30] = "";
	long number;
	int remainder, base = 2;
	scanf("%d", &number);
	if(number == 0) sprintf(answer, "0");
	while(number > 0){
		remainder = number % base;
		number = number / base;
		strcpy(buffer, answer);
		sprintf(answer, "%c%s", table[remainder], buffer);
	}
	printf("%s", answer);
	return 0;
}
